### ISSUE TITLE
Confusing error message

### ISSUE THREAD

**Issue by benwilson512** [absinthe-graphql/absinthe#116]
Given a schema
```
object :item do
 field :id, :id
end
field :foo, :item
```
and a query
```
foo
```
You'll get an error message:
```
"Type `': \"Not found in the schema\""
```

This is because the document is asking for the object itself, instead of one of its fields. What the document SHOULD do is
```
foo {
 id
}
```

so that the leaves are scalars / enums.

We need to return a better error.

**Comment by scottschulthess** [comment]
+1

**Comment by bruce** [comment]
Adding this to the v1.2 milestone. This is directly addressed by the ScalarLeafs validation rule.

**Comment by bruce** [comment]
This also popped up in #161, and is addressed in v1.2. Closing this issue as it's solved in our current development branch. We'd accept a PR to add an error to the v1.1 branch (currently master).
